Shared/Samples/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationView.swift

```diff
@@ -0,0 +1,168 @@
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct EditFeaturesWithFeatureLinkedAnnotationView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The asynchronous action currently being preformed.
+    @State private var selectedAction = AsyncAction.setUpMap
+    
+    /// The current instruction being displayed at the top of the screen.
+    @State private var instruction = Instruction.selectFeature
+    
+    /// The point on the map where the user tapped.
+    @State private var tapLocation: Point?
+    
+    /// The building number of the selected feature.
+    @State private var buildingNumber: Int32?
+    
+    /// The street name of the selected feature.
+    @State private var streetName: String = ""
+    
+    /// A Boolean value indicating whether the edit address alert is presented.
+    @State private var editAddressAlertIsPresented = false
+    
+    /// A Boolean value indicating whether the move confirmation alert is presented.
+    @State private var moveConfirmationAlertIsPresented = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        MapViewReader { mapViewProxy in
+            MapView(map: model.map)
+                .onSingleTapGesture { screenPoint, mapPoint in
+                    if model.selectedFeature == nil {
+                        // Selects a feature at the tap location if there isn't one selected.
+                        selectedAction = .selectFeature(screenPoint: screenPoint)
+                    } else {
+                        // Shows the move confirmation alert if there is already a selected feature.
+                        tapLocation = mapPoint
+                        moveConfirmationAlertIsPresented = true
+                    }
+                }
+                .task(id: selectedAction) {
+                    do {
+                        // Performs the selected action.
+                        switch selectedAction {
+                        case .setUpMap:
+                            try await model.setUpMap()
+                        case .selectFeature(let screenPoint):
+                            let layerIdentifyResults = try await mapViewProxy.identifyLayers(
+                                screenPoint: screenPoint,
+                                tolerance: 10
+                            )
+                            model.selectFirstFeature(from: layerIdentifyResults)
+                            instruction = .moveFeature
```

Comment:
The state machine is not fully correct. When I tap on a blank spot on the map, the instruction still change to `.moveFeature` rather than staying in `.selectFeature`.